update misc/mo-algo-intro (#68)
OI-wiki/i18n: docs/misc/mo-algo-intro.md

author: StudyingFather, Backl1ght, countercurrent-time, Ir1d, greyqz, MicDZ, ouuan

> Mo's algorithm is an algorithm proposed by [Mo Tao](https://www.zhihu.com/people/mythly) . Before him, the Mo's algorithm has been circulating in a small group on Codeforces, but Mo was the first person to ever summarize the algorithm in detail. When Mo proposed the Mo's algorithm, he only analyzed the ordinary version, but after the countless cpntributions from many OIers and ACMers, Mo's algorithm now has a variety of expanded versions.

Reviewer: - circulating -> using/spreading
- When Mo proposed that algorithm, 
- contribution typo

> The Mo's algorithm can solve a type of offline interval query problems and its application is extremely wide. At the same time, it can be easily expanded to handle queries on tree paths and support modification operations.

Reviewer: ```suggestion
The Mo's algorithm can solve a series of offline interval query problems and its applications are extremely wide. At the same time, it can be easily expanded to handle queries on tree paths and support modification operations.
```

> The Mo's algorithm can solve a type of offline interval query problems and its application is extremely wide. At the same time, it can be easily expanded to handle queries on the tree path and support modification operations.

Reviewer: ```suggestion
The Mo's algorithm can solve a type of offline interval query problems and its application is extremely wide. At the same time, it can be easily expanded to handle queries on tree paths and support modification operations.
```
